src/components/EventDetails/EventDetails.js

```diff
@@ -325,32 +359,43 @@ export class EventDetail extends React.Component<Props, State> {
     if (this.timeout) clearTimeout(this.timeout);
   }
 
-  findTxInfo = () => {
-    const { history, event } = this.props;
-    return findTransactionAcrossAccounts(history, event.hash) || {};
-  }
+  findTxInfo = (isCollectible?: boolean) => {
+    const { collectiblesHistory, history, event } = this.props;
 
-  syncEnsRegistry = (txInfo: Transaction) => {
+    if (isCollectible) {
+      return collectiblesHistory.find(({ hash }) => hash === event.hash);
+    }
+
+    return findTransactionAcrossAccounts(history, event.hash);
+  };
+
+  syncEnsRegistry = (txInfo: Transaction | CollectibleTrx) => {
     const { ensRegistry, lookupAddress } = this.props;
     const relatedAddress = this.getRelevantAddress(txInfo);
 
     if (!ensRegistry[relatedAddress]) {
       lookupAddress(relatedAddress);
     }
-  }
+  };
 
-  syncTxStatus = (txInfo: Transaction) => {
-    const {
-      event,
-      updateTransactionStatus,
-    } = this.props;
+  syncTxStatus = (txInfo: Transaction | CollectibleTrx) => {
     if (txInfo.status === TX_PENDING_STATUS) {
-      this.timeout = setTimeout(() => updateTransactionStatus(event.hash), 500);
-      this.timer = setInterval(() => updateTransactionStatus(event.hash), 10000);
+      this.timeout = setTimeout(this.updateTransaction, 500);
+      this.timer = setInterval(this.updateTransaction, 10000);
```

Comment:
this will be called every 10 seconds, what if the user will close and open another transaction during that time?

Reply:
We are calling clear method that cleans Timeout and Interval whenever modal is closed:
https://github.com/pillarwallet/pillarwallet/pull/2021/files#diff-ec904b6cf8e8afec5137f06ae6e584f5R344

So if user would open new modal, timeout and timer will be reset (upon close) and then reset into new value (upon open) :)

Reply:
Yep, you're right!